components/radio/src/auro-radio-group.js

```diff
@@ -216,20 +216,19 @@ export class AuroRadioGroup extends LitElement {
   }
 
   /**
-   * Method for a total reset of the radio element.
+   * Resets component to initial state.
    * @returns {void}
    */
   reset() {
-    this.value = undefined;
     this.index = 0;
```

Comment:
Add some sort of comment that explains that `this.index` is a representation of which radio gets focus